tests/latest/invalid/dbt_yml_files.yml

```diff
@@ -0,0 +1,67 @@
+# this file was generated with dbt init with dbt 1.2.1
+
+version: 2
```

Comment:
can we add an invalid unit test here?

Reply:
We can... but I chose not to because adding an additional invalid item to an already-invalid test doesn't actually test anything. if I added a totally correct unit test block, the CI would still mark it as Task Failed Successfully because there are other failing components. So we probably need to think a bit about what we do there. 

Maybe check for an expected number of failures?

Reply:
OK I did some more digging here, and it's possible to output the list of [`--all-errors`](https://github.com/ajv-validator/ajv-cli?tab=readme-ov-file#test-validation-result:~:text=%24data%20references-,%2D%2Dall%2Derrors,-collect%20all%20validation) which I thought I could just count. But there can be multiple failure entries for a single line of yaml like this: 
```
  {
    instancePath: '/tests/test/+enabled',
    schemaPath: '#/$defs/jinja_string/pattern',
    keyword: 'pattern',
    params: { pattern: '\\{\\{.*\\}\\}' },
    message: 'must match pattern "\\{\\{.*\\}\\}"'
  },
  {
    instancePath: '/tests/test/+enabled',
    schemaPath: '#/oneOf/1/type',
    keyword: 'type',
    params: { type: 'boolean' },
    message: 'must be boolean'
  },
  {
    instancePath: '/tests/test/+enabled',
    schemaPath: '#/oneOf',
    keyword: 'oneOf',
    params: { passingSchemas: null },
    message: 'must match exactly one schema in oneOf'
  },
```

So my new theory for how we could approach this is to generate the failure list manually, and then compare against that in CI. I'm going to punt on that for another day, but I will add a failing mode so that when I come to it at least there's the necessary info.

Reply:
this would be an excellent enhancement! but agree that we should do it outside the scope of this PR